plugins/battery/battery.plugin.zsh

```diff
@@ -110,13 +110,19 @@ elif [[ "$OSTYPE" = freebsd* ]]; then
 elif [[ "$OSTYPE" = linux*  ]]; then
 
   function battery_is_charging() {
-    ! acpi 2>/dev/null | command grep -v "rate information unavailable" | command grep -q '^Battery.*Discharging'
+    if (( $+commands[acpitool] )); then
+      ! apcitool 2>/dev/null | command grep -q '^\s+Battery.*Discharging'
+    else if (( $+commands[acpi] )); then
+      ! acpi 2>/dev/null | command grep -v "rate information unavailable" | command grep -q '^Battery.*Discharging'
+    fi; fi
   }
 
   function battery_pct() {
-    if (( $+commands[acpi] )); then
+    if (( $+commands[acpitool] )); then
+      acpitool 2>/dev/null | command grep -E '^\s+All batteries\s*:' | cut -f1 -d',' | tr -cd '[[:digit:].]'
```

Comment:
This doesn't work if there's only one battery.

Reply:
Do you have an example of that output so I can have a bit to test with? I don't have easy access to one with only one battery right now.

Reply:
I've updated it to try to better handle a single battery situation, let me know how it works